dashboard/src/features/shared/actions/form.js

```diff
@@ -34,10 +31,27 @@ export default function(type, options = {}) {
         data[fieldName] = parseInt(data[fieldName])
       })
 
-      return function(dispatch) {
-        let object = new chain[className](data)
+      const xpubs = data.xpubs || []
+      xpubs.map(key => {
+        if (key.type == 'generate') {
+          promise = promise
+            .then(() => {
+              const alias = (key.value || '').trim()
+                ? key.value.trim()
+                : (data.alias || 'generated') + '-' + uuid.v4()
 
-        return object.create(context())
+              return new chain.MockHsm({alias}).create(context())
+            }).then(newKey => {
+              data.root_xpubs = [...(data.root_xpubs || []), newKey.xpub]
```

Comment:
Might as well initialize `data.root_xpubs` above the loop, where `const xpubs` is getting initialized. That will clean up the append statements here.